testprojects/src/python/nested_runs/run_pants_with_pantsd.py

```diff
@@ -0,0 +1,18 @@
+import os
+import subprocess
+import sys
+
+
+def main():
+  workdir = sys.argv[1]
+  config = os.path.join(workdir, 'pants.ini')
+
+  cmd = './pants --no-pantsrc --pants-config-files={config} --print-exception-stacktrace=True --pants-workdir={workdir} goals'.format(
+    config=config,
+    workdir=workdir,
+  )
+  print(cmd)
+  subprocess.run(cmd, shell=True, check=True)
```

Comment:
Bad to use `shell=True`. This is a security risk and it escapes things in ways we don't want. See https://bandit.readthedocs.io/en/latest/plugins/b602_subprocess_popen_with_shell_equals_true.html.

Instead, remove `shell=True` and convert the command to a list `['./pants', '--no-pantsrc']` etc.

--

Also, you're using `subprocess.run()` which is Py3.5+ only. Because this is a testproject, I think it is totally valid to not support Py2! Just add `compatibility=['CPython>=3.6,<4']` to the BUILD file.

And you should also use f-strings and `pathlib` :) For pathlib, it becomes `pathlib.Path(workdir) / 'pants.ini'`, which  if you think about it is like you're typing the `/` on the command line to go within a folder.

Reply:
Thanks!